src/components/Toggle/Toggle.test.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { axe } from "jest-axe";
+import renderer from "react-test-renderer";
+import * as generateUUID from "../../helpers/generateUUID";
+import Toggle from "./Toggle";
+
+describe("Toggle Accessibility", () => {
+  it("Passes axe accessibility test", async () => {
+    const { container } = render(
+      <Toggle id="inputID" onChange={jest.fn()} labelText="Test Label" />
+    );
+    expect(await axe(container)).toHaveNoViolations();
+  });
+});
+
+describe("Toggle", () => {
+  let changeHandler;
+  let generateUUIDSpy;
+
+  beforeEach(() => {
+    changeHandler = jest.fn();
+  });
+
+  it("Renders with a Toggle input and label", () => {
+    render(<Toggle id="inputID" labelText="Test Label" />);
+    expect(screen.getByLabelText("Test Label")).toBeInTheDocument();
+    expect(screen.getByRole("checkbox")).toBeInTheDocument();
+  });
+
+  it("Renders visible helper or error text", () => {
+    const { rerender } = render(
+      <Toggle
+        id="inputID"
+        labelText="Test Label"
+        helperText="This is the helper text."
+        invalidText="This is the error text :("
+      />
+    );
+    expect(screen.getByText("This is the helper text.")).toBeVisible();
+    expect(
+      screen.queryByText("This is the error text :(")
+    ).not.toBeInTheDocument();
+
+    rerender(
+      <Toggle
+        id="inputID"
+        labelText="Test Label"
+        isInvalid
+        helperText="This is the helper text."
+        invalidText="This is the error text :("
+      />
+    );
+    expect(screen.getByText("This is the error text :(")).toBeVisible();
+    expect(
+      screen.queryByText("This is the helper text.")
+    ).not.toBeInTheDocument();
+  });
+
+  it("Sets the Toggle's ID", () => {
+    render(<Toggle id="inputID" labelText="Test Label" />);
+    expect(screen.getByRole("checkbox")).toHaveAttribute("id", "inputID");
+  });
+
+  it("Calls the UUID generation function if no id prop value is passed", () => {
+    generateUUIDSpy = jest.spyOn(generateUUID, "default");
+    expect(generateUUIDSpy).toHaveBeenCalledTimes(0);
+    render(<Toggle labelText="Test Label" />);
+    expect(generateUUIDSpy).toHaveBeenCalledTimes(2);
```

Comment:
I can't figure out why this was called twice - it makes me nervous that I made a mistake and it's rendering twice.  Thoughts?

Reply:
Every DS component calls the `generateUUID` function unless an `id` is passed to it. In this case, it's the `Label` component that needs an explicit `id` or else it'll call this function. You _can_ make it called twice and add a comment that the second call is from the `Label` or you can pass an `id` value similar to the value for the `HelperErrorText` component.